main.bicep

```diff
@@ -450,6 +450,166 @@ For more information on the telemetry collected by this module, that is controll
 ''')
 param disableTelemetry bool = false
 
+@sys.description('Guid for the deployment script resources names based on subscription Id.')
+var deploymentScriptResourcesSubGuid = substring((subscriptionAliasEnabled && empty(existingSubscriptionId)) ? createSubscription.outputs.subscriptionId : existingSubscriptionId,0,8)
+
+@sys.description('The name of the resource group to create the deployment script for resource providers registration.')
+param deploymentScriptResourceGroupName string = 'rsg-${deployment().location}-ds'
+
+@sys.description('The name of the deployment script to register resource providers')
+param deploymentScriptName string = 'ds-${deployment().location}'
+
+@sys.description('The name of the user managed identity for the resource providers registration deployment script.')
+param deploymentScriptManagedIdentityName string = 'id-${deployment().location}'
+
+
+@metadata({
+  example: {
+    'Microsoft.Compute' : ['InGuestHotPatchVMPreview']
+    'Microsoft.Storage' : []
+  }
+
+})
+@sys.description('''
+An object of resource providers and resource providers features to register. If left blank/empty, no resource providers or features will be registered.
```

Comment:
The "If left blank/empty" part of the description is not true, can we remove or reword?

We should register the list of providers by default